lib/charms/grafana_agent/v0/cos_agent.py

```diff
@@ -758,6 +758,13 @@ def _dashboards(self) -> List[str]:
                 # because there is currently no other way to communicate the dashboard path separately.
                 # https://github.com/canonical/grafana-k8s-operator/pull/363
                 dashboard["uid"] = DashboardPath40UID.generate(self._charm.meta.name, rel_path)
+
+                # Add tags
+                tags: List[str] = dashboard.get("tags", [])
+                if not any(tag.startswith("charm: ") for tag in tags):
+                    tags.append(f"charm: {self._charm.meta.name}")
+                dashboard["tags"] = tags
```

Comment:
The `_dashboards` property is too much nested... This block is adding one more indentation level. Maybe we can extract this to a new method, something like this:

```python
                dashboard["tags"] = self._dashboard_tags(dashboard)
                ...


         def _dashboard_tags(self, dashboard):
             tags: List[str] = dashboard.get("tags", [])
             if not any(tag.startswith("charm: ") for tag in tags):
                 tags.append(f"charm: {self._charm.meta.name}")
             return tags
```